src/App.jsx

```diff
@@ -14,45 +16,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  let allGoods;
+
+  switch (sortField) {
+    case 'alphabet':
+      allGoods = [...goods].sort();
+      break;
+    case 'length':
+      allGoods = [...goods].sort((good1, good2) => good1.length - good2.length);
+      break;
+    case '':
+    default:
+      allGoods = [...goods];
+      break;
+  }
+
+  if (isReversed) {
+    allGoods.reverse();
+  }
+
+  return allGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== 'alphabet',
+          })}
+          onClick={() => setSortField('alphabet')}
```

Comment:
```suggestion
          onClick={sortAlphabetically}
```
It's better to create separate functions to handle events. This way, your code will be easier to maintain, extend, and reuse.

Imagine you'll have a situation where you need to reuse the same functionality on another event. Then you'd be able to just use the same function you've created. Also, if you need to extend the functionality of event handling in the future, it'd be easier to do so if you have a separate function. Moreover, if this function's used more than once you'll need to add new functionality only in one function and then call it wherever you need to.